### ISSUE TITLE
Support for batch indexing

### ISSUE THREAD

**Issue by dvirsky** [ethanhann/redisearch-php#1]
The API of your client looks very very nice.
One thing that's missing IMO is the possibility to use a pipeline to index multiple documents at a time. This can significantly reduce indexing time. 

See the Python implementation here: https://github.com/RedisLabs/redisearch-py/blob/master/redisearch/client.py#L69

**Comment by ethanhann** [comment]
I was thinking about that. I'll take a look.

**Comment by ethanhann** [comment]
I added support for batch indexing with an addMany() method that takes an array of documents. It wraps some simple multi/exec logic. I'm mulling over some chunking ideas that might fit well with some popular frameworks.

Doc: http://www.ethanhann.com/redisearch-php/indexing/#batch-indexing

I ran some simple performance tests. The phpredis extension showed good results. **addMany** was about 25% faster than **add** with the default Redis::MULTI option. Using Redis::PIPELINE **addMany** was twice as fast.

Using the Predis client library **addMany** was surprisingly about 7% slower. Either I am doing something wrong (which seems unlikely because the code is simple) or there is a bug in Predis. I'm going to look into this before closing this issue.

**Comment by ethanhann** [comment]
Apparently the Predis performance issue is a limitation of the library. The Predis developers recommend installing [phpiredis](https://github.com/nrk/phpiredis) (which is different from phpredis - note the additional "i") to improve multi performance. **addMany** is consistently about 10-15% faster than **add* when using Predis+phpiredis. 

tl;dr Use phpredis instead of Predis if bulk indexing performance (or performance in general) really matters to you.
